### DISCUSSION TITLE
2FA doesnt work

### DISCUSSION THREAD

**Discussion by geoffw123** [community/community#63788] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

First point. I do not want to go to all this hassle of using 2FA. I was perfectly happy with the way it used to be prior to your decision to change this policy.

Now I am forced to use 2FA , I have just wasted 1 hour trying to get it working. I have selected use 2FA with mobile phone sms code. I have tried 4 or 5 times to set it up, I passed the human check, I then entered my country of UK, i then typed in my working fine mobile phone number.  I have received 0 sms messages from Github to enable me to continue this process.

I give up, so I am sending you this feedback. I will try again in a few days time to see if you have fixed it.  Otherwise if you dont want me to use github, I am perfectly happy to delete all my code I uploaded

**Comment by hexaheximal** [comment]
If you end up leaving GitHub (which you really should, see https://nogithub.codeberg.page/), you should migrate to codeberg, which is what I'm also migrating to. They are a non-profit organization (unlike GitHub, which is literally owned by Microsoft) and codeberg itself is FLOSS. (unlike GitHub, which is proprietary)
Escape before further implosion.

**Comment by mehdi-najaran** [comment]
which country are you ؟
My suggestion is to use the application
Related to Google, it has the least bugs and disturbances

The SMS system sometimes has problems, as a result, the GitHub account may have problems

Also, be sure to save the backup code

**Reply by geoffw123** [reply]
As I said in my original email I am in the UK
I am aware of the ethical issues that Microsoft have ignored using Open source code for their AI products.
I am currently more hacked off that they have removed my free choice as to whether I want to use 2FA or not, and it is being imposed on me.
I know my phone and SMS is working fine as I used an SMS code yesterday to validate a Tesco shopping account.
Kind of amusing and embarassing for Microsoft as Tesco can implement a working and reliable SMS verification system and MIcrosoft cant :)
